master/internal/resourcemanagers/job.go

```diff
@@ -0,0 +1,92 @@
+package resourcemanagers
+
+import (
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/jobv1"
+)
+
+// FIXME haven't decided if resource manager actor should be responsible for this or not
+// we don't want a separate actor do we? could be useful for streaming job endpoints
+type GetJobOrder struct{}
+
+func filterAllocateRequests(reqs AllocReqs) AllocReqs {
+	isAdded := make(map[model.JobID]bool)
+	rv := make(AllocReqs, 0)
+	for _, req := range reqs {
+		job := req.Job
+		if job == nil {
+			continue
+		} else if _, ok := isAdded[job.JobID]; ok {
+			continue
+		}
+		isAdded[job.JobID] = true
+		rv = append(rv, req)
+	}
+	return rv
+}
+
+func allocReqsToJobSummaries(reqs AllocReqs) (summaries []*sproto.JobSummary) {
+	for _, req := range filterAllocateRequests(reqs) {
+		summaries = append(summaries, req.Job)
+	}
+	return summaries
+}
+
+// allocReqsToJobOrder convertes sorted allocation requests to job order.
+func allocReqsToJobOrder(reqs []*sproto.AllocateRequest) (jobIds []string) {
+	for _, req := range filterAllocateRequests(reqs) {
```

Comment:
Is `filterAllocateRequests` supposed to be sorted?

Reply:
yeah. that func expects a sorted list of allocations, (and outputs a sorted one) but has no way of checking it. maybe I should clarify this somewhere.

Reply:
any suggestions?

Reply:
hmmm maybe I'm not reading this correctly, but it doesn't look to me like we can assume that the returned req list is sorted